### ISSUE TITLE
Erroneous PreState creation in AbsInt Hoare-Triple Check

### ISSUE THREAD

**Issue by greitsch** [ultimate-pa/ultimate#157]
F: examples/svcomp/eca-rers2012/Problem01_label00_true-unreach-call.c
S: ai/svcomp-Reach-32bit-Automizer_Default+AIv2_CON_Debug.epf
T: AutomizerC.xml

For Settings: It is preferable to enable TraceAbstraction DEBUG output to see all output.

During weakening of the interpolant sequence, the state

`{~a12 = 8; ~a21 = 1; ~a17 = 1; ~a16 = 5; ~a8 = 15; ~a7 = 0; ~a20 = 1; calculate_output_#in~input = 1Z \ {0}; TOP: old(~a12), old(~a21), old(~a17), old(~a16), calculate_output_~input, calculate_output_#res, old(~a8), old(~a7), old(~a20)}`

is reduced to

`{calculate_output_#in~input = 1Z \ {0};}`

for transition

`~input := #in~input;`

The Hoare-triple to be checked therefore is

```
{calculate_output_#in~input = 1Z \ {0};}

~input := #in~input;

{~a12 = 8; ~a21 = 1; ~a17 = 1; ~a16 = 5; ~a8 = 15; ~a7 = 0; calculate_output_~input = 1Z \ {0}; ~a20 = 1; }
```

However, the Hoare-triple checker seems to construct an erroneous Pre-State. It will check the following Hoare-triple:

```
{calculate_output_#in~input = 1Z \ {0}; TOP: ~a12, old(~a12), ~a21, old(~a21), ~a17, old(~a17), ~a16, old(~a16), calculate_output_~input, calculate_output_#res, ~a8, old(~a8), ~a7, old(~a7), ~a20, old(~a20) }

~input := #in~input;

{~a12 = 8; ~a21 = 1; ~a17 = 1; ~a16 = 5; ~a8 = 15; ~a7 = 0; calculate_output_~input = 1Z \ {0}; ~a20 = 1; }
```

for which the SMT solver says is not inductive.

**Comment by greitsch** [comment]
Won't fix. The problem is a different one.

**Comment by greitsch** [comment]
This issue was resolved due to the fixes in 49d1309.
